Userland/Applications/Calculator/main.cpp

```diff
@@ -20,10 +20,11 @@
 
 ErrorOr<int> serenity_main(Main::Arguments arguments)
 {
-    TRY(Core::System::pledge("stdio recvfd sendfd rpath unix"));
+    using enum Kernel::Pledge;
+    TRY((Core::System::Promise<stdio, Kernel::Pledge::recvfd, Kernel::Pledge::sendfd, rpath, unix>::pledge()));
     auto app = TRY(GUI::Application::try_create(arguments));
 
-    TRY(Core::System::pledge("stdio recvfd sendfd rpath"));
+    TRY((Core::System::Promise<stdio, Kernel::Pledge::recvfd, Kernel::Pledge::sendfd, rpath>::pledge()));
```

Comment:
This is a huge regression in readability. Are we really succeeding at something here?

Reply:
Honestly, at this point I don't feel like there's any improvement I can make that will not be disliked by someone. I agree that some pledges are less readable than before, but most are not. (The explicit namespaces are needed for name conflict reasons btw.) Actual `|` has been voted down by @bgianfo, @trflynn89 made part of this one's templates which I like but @alimpfard would like to have even more template expressiveness like in https://discord.com/channels/830522505605283862/834391539425476680/952899712313360454 . The builder pattern has also been discussed though that's out of the window in my opinion.

So I'm ready to give up here. The solution we have right now is not good, and we all seem to agree on that point. But we don't seem to agree on what is a better solution. I think that from a security standpoint this is still massively better than before, but at what cost. I'm now tired of getting told again and again by different people that what I have is not good, only for me to go and change it and get told by someone else that it's still not good. We seem unable to settle on anything.

I won't close this PR, especially because either way I don't have time to work on this until the weekend. If someone can come up with something that everyone likes before that, it'd be great, I'll hand this project over immediately if they even make a PR (and if not I'll implement their solution). If not, I'll close this as "no consensus formed" and we'll live with strings.

Reply:
I'm sorry about the round-and-round, but you're changing the first line of every single program in the system. Some bikeshedding tolerance is required in situations like that. :)

So I'll ask @bgianfo @trflynn89 @alimpfard, compared to the extremely simple and readable interface we have today, do you think `Kernel::Pledge::This`, `Kernel::Pledge::Is`, `Kernel::Pledge::An`, `Kernel::Pledge::Improvement`, or is there perhaps some way we retain the concise style of the current API?

Reply:
Yeah the idea was to actively avoid `Kernel::Pledge::` mayhem so that the call sites would look like:
```c++
TRY(Core::System::Promise<stdio, rpath, tty>::pledge());
```

But it sounds like the only reason this isn't possible as-is with *every* pledge is that there are some naming conflicts (e.g. `recvfd`)? In that case, we could redefine the enum to use our normal enum casing:
```c++
TRY(Core::System::Promise<RecvFD, SendFD>::pledge());
```

or whatever.

That said, I don't get any such conflict errors when I build this branch locally and remove `Kernel::Pledge::` from the pledges like `recvfd`, even if I directly include `<sys/socket.h>` where they are defined. Works fine for both clang and gcc on godbolt as well: https://godbolt.org/z/Tjr17nfWW
e.g. in Audio/main.cpp I'm able to do this without issue, and that seems readable to me:
```c++
TRY((Core::System::Promise<stdio, recvfd, sendfd, rpath, wpath, cpath, unix>::pledge()));
```

Reply:
Okay, maybe that was a false positive from clangd.